Thanks @cbm755, I think it works great!
```
In [19]: print pretty(sqrt(2*5*6*7*878787*54654654)*x, unicode_square_roots=False)
      _______________  
126⋅╲╱ 1270629614410 ⋅x

In [20]: print pretty(sqrt(2*5*6*7*878787*54654654)*x)
126⋅√1270629614410⋅x
```
I think this way it is safe --- if it causes problems, people can turn it off immediately for the given expression. I think the way it looks is better, so it should be on by default.

However, you have a test failure there:
```
sympy/printing/pretty/tests/test_pretty.py[0] Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/travis/virtualenv/python3.2.5/lib/python3.2/site-packages/sympy-0.7.6_git-py3.2.egg/sympy/utilities/runtests.py", line 512, in _test
    slow=slow, enhance_asserts=enhance_asserts))
  File "/home/travis/virtualenv/python3.2.5/lib/python3.2/site-packages/sympy-0.7.6_git-py3.2.egg/sympy/utilities/runtests.py", line 964, in test
    self.test_file(f, sort, timeout, slow, enhance_asserts)
  File "/home/travis/virtualenv/python3.2.5/lib/python3.2/site-packages/sympy-0.7.6_git-py3.2.egg/sympy/utilities/runtests.py", line 1028, in test_file
    code = compile(source, filename, "exec")
  File "/home/travis/virtualenv/python3.2.5/lib/python3.2/site-packages/sympy-0.7.6_git-py3.2.egg/sympy/printing/pretty/tests/test_pretty.py", line 1719
    """
      ^
SyntaxError: invalid syntax
Traceback (most recent call last):
  File "<stdin>", line 3, in <module>
Exception: Tests failed
```
This happens in Python 3.2 only and the reason is that it doesn't support the `u` in the string --- you need to use `u()`, see the other tests.

--- a/sympy/printing/pretty/pretty.py
+++ b/sympy/printing/pretty/pretty.py
@@ -38,7 +38,7 @@ class PrettyPrinter(Printer):
         "use_unicode": None,
         "wrap_line": True,
         "num_columns": None,
-        "unicode_square_roots": True,
+        "use_unicode_sqrt_char": True,
     }
 
     def __init__(self, settings=None):
@@ -1330,7 +1330,7 @@ class PrettyPrinter(Printer):
         bpretty = self._print(base)
 
         # In very simple cases, use a single-char root sign
-        if (self._settings['unicode_square_roots'] and self._use_unicode
+        if (self._settings['use_unicode_sqrt_char'] and self._use_unicode
             and expt is S.Half and bpretty.height() == 1
             and (bpretty.width() == 1
                  or (base.is_Integer and base.is_nonnegative))):
@@ -1962,7 +1962,7 @@ def pretty_print(expr, **settings):
         use full precision. Default to "auto"
     order : bool or string, optional
         set to 'none' for long expressions if slow; default is None
-    unicode_square_roots : bool, optional
+    use_unicode_sqrt_char : bool, optional
         use compact single-character square root symbol (when unambiguous);
         default is True.
 

--- a/sympy/printing/pretty/tests/test_pretty.py
+++ b/sympy/printing/pretty/tests/test_pretty.py
@@ -1709,22 +1709,6 @@ u("√2")
     assert pretty(expr) == ascii_str
     assert upretty(expr) == ucode_str
 
-
-def test_pretty_sqrt_symbol_knob():
-    expr = sqrt(2)
-    ucode_str1 = \
-u"""\
-  ___\n\
-╲╱ 2 \
-"""
-    ucode_str2 = \
-u("√2")
-    assert xpretty(expr, use_unicode=True,
-                   unicode_square_roots=False) == ucode_str1
-    assert xpretty(expr, use_unicode=True,
-                   unicode_square_roots=True) == ucode_str2
-
-
     expr = 2**Rational(1, 3)
     ascii_str = \
 """\
@@ -1837,18 +1821,31 @@ u("""\
     assert pretty(expr) == ascii_str
     assert upretty(expr) == ucode_str
 
+
+def test_pretty_sqrt_char_knob():
+    # See PR #9234.
+    expr = sqrt(2)
+    ucode_str1 = \
+u("""\
+  ___\n\
+╲╱ 2 \
+""")
+    ucode_str2 = \
+u("√2")
+    assert xpretty(expr, use_unicode=True,
+                   use_unicode_sqrt_char=False) == ucode_str1
+    assert xpretty(expr, use_unicode=True,
+                   use_unicode_sqrt_char=True) == ucode_str2
+
+
+def test_pretty_sqrt_longsymbol_no_sqrt_char():
+    # Do not use unicode sqrt char for long symbols (see PR #9234).
     expr = sqrt(Symbol('C1'))
-    ascii_str = \
-"""\
-  ____\n\
-\/ C1 \
-"""
     ucode_str = \
 u("""\
   ____\n\
 ╲╱ C₁ \
 """)
-    assert pretty(expr) == ascii_str
     assert upretty(expr) == ucode_str
 
 

